docs/build/README.md

```diff
@@ -1,89 +1,109 @@
-## How does Building Garden Linux work?
 
-By running the build.sh command, you start the Build Process.
+**Table of Content**
+- [Introduction](#introduction)
+	- [Build via make (recommended)](#build-via-make-recommended)
+	- [Build via build.sh](#build-via-buildsh)
+	- [Build Artifacts](#build-artifacts)
+- [Kernel Module](#kernel-module)
+	- [build-kernelmodule container](#build-kernelmodule-container)
+- [Package Build](#package-build)
+	- [Git Source](#git-source)
+	- [Snapshot Source](#snapshot-source)
+
+# Introduction
+
+Garden Linux makes use of the Github Release feature. Checkout available Releases [here](https://github.com/gardenlinux/gardenlinux/releases).
+But you can also decide to customize and build Garden Linux with the included build tools. 
+
+Customization is done via the the Garden Linux Feature System. 
+The feature system allows for a deliberate, and granular selection of features to be included in the final Garden Linux Image.
+
+## Build via make (recommended)
+Use the [Makefile](/Makefile) to build a Garden Linux Image with a pre-defined set of features. 
+```
+# Example 
+make metal-dev
+```
+For more targets, checkout the [Makefile](/Makefile). The targets call ` build.sh` with a pre-defined set of features.  
+
+You can also customize a Makefile target to your needs, e.g. by adding a feature.
+
+## Build via build.sh 
+If you really want to directly call build.sh, you can checkout [Makefile](/Makefile) for some good examples and start from there.
+
+In general, the `build.sh` starts and prepares the build container, in which `garden-build` and `garden-test` are executed.
+
+## Build Artifacts
+Build artifacts are stored in the output folder (default `.build/`).
+Some artifacts will only be created by certain features. 
+
+# Kernel Module
+Drivers/LKMs not included in upstream linux of kernel.org can be build out of tree.
+
+## build-kernelmodule container 
+We provide build container that come with gardenlinux linux-headers installed. 
+These build containers have a `uname -r` wrapper installed. 
+This wrapper outputs the latest installed kernel header in that container.
+
+Container is created here: https://gitlab.com/gardenlinux/driver/gardenlinux-driver-build-container
+
+```
+docker pull registry.gitlab.com/gardenlinux/driver/gardenlinux-driver-build-container/gl-driver-build:dev
+```
+
+1. Load your kernel module sources into the container
```

Comment:
In .md files, for a numbered list, prefix all the bullet points with just `1.` and leave the numbering to the markdown renderer.